src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/PerformerNameAndAddress.java

```diff
@@ -0,0 +1,103 @@
+
+/**
+ * Example: NAD+SLA+A2442389:902++DR J SMITH'<br>
+ * Example: NAD+SLA+++ST JAMES?'S UNIVERSITY HOSPITAL'
+ */
+@Getter
+@Setter
+@Builder
+@RequiredArgsConstructor
+public class PerformerNameAndAddress extends Segment {
+
+    private static final String KEY = "NAD";
+    private static final String QUALIFIER = "SLA";
+    public static final String KEY_QUALIFIER = KEY + "+" + QUALIFIER;
+    private static final int PERFORMING_NAME_INDEX_IN_EDIFACT_STRING = 4;
+    private static final int PERFORMER_ID_INDEX_IN_EDIFACT_STRING = 2;
+    private static final int PERFORMER_CODE_INDEX_IN_EDIFACT_STRING = 3;
+
+    @NonNull
+    private final String identifier;
+    private final HealthcareRegistrationIdentificationCode code;
+    @NonNull
+    private final String performingOrganisationName;
+    @NonNull
+    private final String partyName;
+
+    public static PerformerNameAndAddress fromString(String edifactString) {
+        if (!edifactString.startsWith(KEY_QUALIFIER)) {
+            throw new IllegalArgumentException(
+                "Can't create " + PerformerNameAndAddress.class.getSimpleName() + " from " + edifactString
+            );
+        }
+
+        String[] keySplit = Split.byPlus(edifactString);
+        String[] colonSplit = Split.byColon(keySplit[2]);
+        String performerID = colonSplit[0];
+
+        if (performerID.isBlank()) {
+            // if identifier is blank - organisation/department
+            String performingOrganisationName = keySplit[PERFORMING_NAME_INDEX_IN_EDIFACT_STRING];
+            return new PerformerNameAndAddress("", null, performingOrganisationName, "");
+        } else {
+            String performerCode = colonSplit[1];
+            String performerPartyName = keySplit[PERFORMING_NAME_INDEX_IN_EDIFACT_STRING];
+            return new PerformerNameAndAddress(
+                performerID,
+                HealthcareRegistrationIdentificationCode.fromCode(performerCode),
+                "",
+                performerPartyName
+            );
+        }
+    }
+
+    @Override
+    public String getKey() {
+        return KEY;
+    }
+
+    @Override
+    public String getValue() {
+        return QUALIFIER
+            + PLUS_SEPARATOR
+            + (identifier.isBlank() ? "" : identifier + COLON_SEPARATOR + code.getCode())
+            + PLUS_SEPARATOR
+            + PLUS_SEPARATOR
+            + (identifier.isBlank() ? performingOrganisationName : partyName);
+    }
+
+    @Override
+    protected void validateStateful() throws EdifactValidationException {
+
+    }
+
+    @Override
+    public void preValidate() throws EdifactValidationException {
+        if (this.identifier.isBlank()) {
+            if (performingOrganisationName.isBlank()) {
+                throw new EdifactValidationException(getKey() + ": Attribute performingOrganisationName is required");
+            }
+        } else {
+            if (code.getCode().isBlank()) {
```

Comment:
`code` is not annotated with `@NonNull`, so if `code == null` this will throw a `NullPointerException` instead of an `EdifactValidationException`.
```suggestion
            if (code == null || code.getCode().isBlank()) {
```

Reply:
Done in 1bbb683